network/memsocket/src/lib.rs

```diff
@@ -292,7 +299,10 @@ impl MemorySocket {
     /// # Ok(())}
     /// ```
     pub fn connect(port: u16) -> Result<MemorySocket> {
-        let mut switchboard = (&*SWITCHBOARD).lock().unwrap();
+        let mut switchboard =  match (&*SWITCHBOARD).lock() {
+            Ok(switchboard) => switchboard,
+            Err(_e) => return Err(ErrorKind::Other.into()),
+        };
```

Comment:
Is there a reason to throw the error away?

Also, what's wrong with panicing on a poisoned lock?  If the lock is poisoned, this whole code basically stops working without manual intervention.

Reply:
We've had this discussion before, but we never came to an actual agreement on if we should panic when locks are poisoned.
My point of view is that: if we are still running at this point, and we didn't crash yet, it's because we decided to not crash the whole process when the relevant thread panicked.
If we made that decision, it's because we wanted the rest of the system to continue working. Perhaps because we might still be able to do useful work. For example, if consensus was running in a thread that crashed, state-sync or mempool could still help the network.
Wdyt?

cc @davidiw @bmwill

Reply:
Yes we should panic on poisoned locks, preferably with `unwrap`. The only time locks are ever poisoned is if a thread panics while still holding the lock. Such cases are not really recoverable (they can be but I guarantee we are not prepared to recover from such cases.

This is one such case where blindly applying the "Don't panic in a function that can return a result" is the wrong thing to do, we do indeed want to panic here.

Reply:
I've reverted the changes for the .lock().unwrap()